subgraphs/ribbon-finance/src/common/initializers.ts

```diff
@@ -0,0 +1,489 @@
+import {
+  Account,
+  FinancialsDailySnapshot,
+  RewardToken,
+  Token,
+  UsageMetricsDailySnapshot,
+  UsageMetricsHourlySnapshot,
+  YieldAggregator,
+  _Auction,
+  _LiquidityGauge,
+  Vault as VaultStore,
+  VaultHourlySnapshot,
+  VaultDailySnapshot,
+  VaultFee,
+  _SwapOffer,
+} from "../../generated/schema";
+import * as utils from "./utils";
+import * as constants from "./constants";
+import { getUsdPricePerToken } from "../prices";
+import {
+  Address,
+  ethereum,
+  BigInt,
+  BigDecimal,
+  dataSource,
+} from "@graphprotocol/graph-ts";
+import { ERC20 as ERC20Contract } from "../../generated/templates/LiquidityGauge/ERC20";
+import { LiquidityGauge as LiquidityGaugeTemplate } from "../../generated/templates";
+import { LiquidityGaugeV5 as GaugeContract } from "../../generated/templates/LiquidityGauge/LiquidityGaugeV5";
+import { RibbonThetaVaultWithSwap as VaultContract } from "../../generated/templates/LiquidityGauge/RibbonThetaVaultWithSwap";
+import { Versions } from "../versions";
+
+export function getOrCreateToken(
+  address: Address,
+  block: ethereum.Block,
+  vault: Address = constants.NULL.TYPE_ADDRESS,
+  isOutputToken: bool = false
+): Token {
+  let token = Token.load(address.toHexString());
+
+  if (!token) {
+    token = new Token(address.toHexString());
+    const contract = ERC20Contract.bind(address);
+    token.name = utils.readValue<string>(contract.try_name(), "");
+    token.symbol = utils.readValue<string>(contract.try_symbol(), "");
+    token.decimals = utils.readValue<i32>(
+      contract.try_decimals(),
+      constants.DEFAULT_DECIMALS.toI32() as u8
+    );
+    token._vaultId = vault.toHexString();
+    token.lastPriceBlockNumber = block.number;
+    if (isOutputToken) {
+      token._isOutputToken = true;
```

Comment:
This is also redundant, you can set `token._isOutputToken = isOutputToken` outside of the if block.

btw, I am just being picky here, but we should aim to make our code as succinct as possible for easier readability and maintenance in the future